src/components/cell-type-external-resource.jsx

```diff
@@ -1,84 +1,38 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
 import { bindActionCreators } from 'redux'
 
-import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
-import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
-import getMuiTheme from 'material-ui/styles/getMuiTheme'
-import CheckCircle from 'material-ui/svg-icons/action/check-circle'
-import ErrorCircle from 'material-ui/svg-icons/alert/error'
-import UnloadedCircle from 'material-ui/svg-icons/content/remove'
-
 import TwoRowCell from './two-row-cell'
 import CellEditor from './cell-editor'
+import ExternalResourceOutputHandler from './output-handler-external-resource'
 
-import actions from '../actions'
 import { getCellById } from '../notebook-utils'
 
-class ExternalResourceCell extends React.Component {
-  outputComponent = () => {
-    if (this.props.value === undefined) return undefined
-    const outs = this.props.value.filter(d => d.src !== '').map((d, i) => {
-      let statusExplanation
-      let statusIcon
-      if (d.status === undefined) {
-        statusIcon = <UnloadedCircle />
-      } else {
-        statusIcon = (d.status === 'loaded'
-          ? <CheckCircle color="lightblue" />
-          : <ErrorCircle color="firebrick" />
-        )
-      }
-      if (Object.prototype.hasOwnProperty.call(d, 'statusExplanation')) {
-        // TODO: Don't use an array index as a key here (See react/no-array-index-key linter)
-        statusExplanation = <div key={i} className="dependency-status-explanation">{d.statusExplanation}</div>  // eslint-disable-line
-      }
-      return (
-        // TODO: Don't use an array index as a key here (See react/no-array-index-key linter)
-        /* eslint-disable */
-        <div className="dependency-container" key={`erc-${this.props.cellId}-${i}`}>
-          <div className="dependency-row">
-            <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
-              {statusIcon}
-            </MuiThemeProvider>
-            <div className="dependency-src">{d.src}</div>
-          </div>
-          {statusExplanation}
-        </div>
-        /* eslint-enable */
-      )
-    })
-    return (
-      <div className="dependency-output">
-        {outs}
-      </div>
-    )
+export class ExternalResourceCellUnconnected extends React.Component {
+  static propTypes = {
+    cellId: PropTypes.number.isRequired,
+    value: PropTypes.array.isRequired,
```

Comment:
i'm getting "warning.js:33 Warning: Failed prop type: The prop `value` is marked as required in `ExternalResourceCellUnconnected`, but its value is `undefined`.";

is this prop not required?

Reply:
i.e., should it just be  `value: PropTypes.array`?

Reply:
I'll remove `.isRequired` from it since it will often be `undefined`. Not sure why I didn't get this error before.